### PR TITLE
Fix #301 (X-address w/ tag in autofill)

### LINKED ISSUE #301: safe_sign_transaction breaks on X-address with omitted destination tag [v1.2.0]
# Summary
The `safe_sign_transaction(payment, wallet)` method should succeed if the payment has a `Destination` that is an X-address with a tag and no `DestinationTag` field (it should fill in the field automatically), but it fails with a KeyError instead.

# Steps to Reproduce

```
>>> import xrpl
>>> w = xrpl.wallet.Wallet.create()
>>> p = xrpl.models.transactions.Payment(account=w.classic_address, sequence=1, amount=xrpl.utils.xrp_to_drops(20), fee="10", flags=0, destination=xrpl.core.addresscodec.classic_address_to_xaddress("rDUNw4eWCFCJpuL4ikbnczYhTh2hZdZ9Uf", tag=1, is_test_network=True))
>>> xrpl.transaction.safe_sign_transaction(p, w)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.9/site-packages/xrpl/transaction/main.py", line 86, in safe_sign_transaction
    return asyncio.run(
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/lib/python3.9/site-packages/xrpl/asyncio/transaction/main.py", line 79, in safe_sign_transaction
    transaction_json = _prepare_transaction(transaction, wallet)
  File "/usr/lib/python3.9/site-packages/xrpl/asyncio/transaction/main.py", line 169, in _prepare_transaction
    _validate_account_xaddress(transaction_json, "Destination", "DestinationTag")
  File "/usr/lib/python3.9/site-packages/xrpl/asyncio/transaction/main.py", line 209, in _validate_account_xaddress
    if json[tag_field] and json[tag_field] != tag:
KeyError: 'DestinationTag'
```

### PR DESCRIPTION
**Pull request by mDuo13**
## High Level Overview of Change

Corrects a simple error that caused KeyError instead of the correct behavior as described in #301.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation Updates (to make flake8 stop complaining)

## Test Plan

I did not figure out how to add unit or integration tests for this but I confirmed that the generated Payment model is correct instead of raising an error with the steps to reproduce in the ticket.

### CODE REVIEW

File: xrpl/asyncio/transaction/main.py [hunk 1]
```diff
@@ -202,19 +202,32 @@ def _validate_account_xaddress(
...
+
+    Raises:
+        XRPLException: if both an X-Address containing a tag and a tag field are
+            provided and they do not match.
     """
     if is_valid_xaddress(json[account_field]):
         account, tag, _ = xaddress_to_classic_address(json[account_field])
         json[account_field] = account
-        if json[tag_field] and json[tag_field] != tag:
+        if tag_field in json and json[tag_field] != tag:
```
Comment (mvadari) [inline xrpl/asyncio/transaction/main.py:218]: Would you mind adding a test case for this fix?

**Review by JST5000 — APPROVED** [review]
LGTM

**Review by khancode — APPROVED** [review]
LGTM

File: xrpl/asyncio/transaction/main.py (hunk 1 above)
Comment (JST5000) [reply xrpl/asyncio/transaction/main.py:229]: Added

**Review by mvadari — APPROVED** [review]

**Pull request merged**